app/code/Magento/InventoryCatalogSearch/Plugin/Search/FilterMapper/AdaptStockStatusFilterPlugin.php

```diff
@@ -189,15 +208,25 @@ function ($fromPart) {
 
     /**
      * @return string
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
      */
     private function getStockTableName(): string
     {
-        $website = $this->storeManager->getWebsite();
-        $stock = $this->stockResolver->execute(
-            SalesChannelInterface::TYPE_WEBSITE,
-            $website->getCode()
-        );
-        $tableName = $this->stockIndexTableNameResolver->execute((int)$stock->getStockId());
+        $tableName = $this->stockIndexTableNameResolver->execute((int)$this->getStockId());
         return $this->resourceConnection->getTableName($tableName);
     }
+
+    /**
+     * @return int
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    private function getStockId()
```

Comment:
in general it does not make sense to extract function for the only function call, it introduces more mess than clarity

Reply:
We need it to avoid code duplication when we check if it's default stock.